aws-frauddetector-detector/src/aws_frauddetector_detector/helpers/update_worker_helpers.py

```diff
@@ -96,31 +96,40 @@ def update_detector_version_for_detector_update(afd_client,
         }
         desired_rules.append(rule_dict)
     if previous_model.DetectorVersionStatus != DRAFT_STATUS:
+        LOG.info(
+            "previous detector version status was not DRAFT. creating a new detector version"
+        )
         api_helpers.call_create_detector_version(
             frauddetector_client=afd_client,
             detector_id=model.DetectorId,
             rules=desired_rules,
             rule_execution_mode=model.RuleExecutionMode,
-            model_versions=None,
-            external_model_endpoints=None,
+            model_versions=[],
+            external_model_endpoints=[],
             detector_version_description=model.Description,
             detector_version_tags=model_helpers.get_tags_from_tag_models(model.Tags))
     else:
+        LOG.info(
+            "previous detector version status was DRAFT. updating detector version in place"
+        )
         api_helpers.call_update_detector_version(
             frauddetector_client=afd_client,
             detector_id=model.DetectorId,
+            detector_version_id=model.DetectorVersionId,
             rules=desired_rules,
             rule_execution_mode=model.RuleExecutionMode,
-            model_versions=None,
-            external_model_endpoints=None,
+            model_versions=[],
+            external_model_endpoints=[],
```

Comment:
These were bugs in DRAFT -> [anything] update: Update DV did not have the proper parameters.